src/builder.rs

```diff
@@ -206,14 +215,39 @@ pub enum InitError<DI, P> {
     /// This error is returned when the configuration passed to the builder is
     /// invalid. For example, when the combination of bit depth and interface
     /// kind isn't supported by the selected model.
-    InvalidConfiguration,
+    InvalidConfiguration(ConfigurationError),
+}
+
+/// Specifics of [InitError::InvalidConfiguration] if configuration was found invalid
+#[non_exhaustive]
+#[derive(Debug)]
+pub enum ConfigurationError {
+    /// Unsupported interface kind.
+    ///
+    /// The chosen interface isn't supported by the selected model. Note that
+    /// some controller models don't support all combinations of physical
+    /// interface and color formats. To resolve this, try to use another color
+    /// format if available (e.g. [`ILI9486Rgb666`](crate::models::ILI9486Rgb666) instead of
+    /// [`ILI9486Rgb565`](crate::models::ILI9486Rgb565) if you use a SPI connection)
+    UnsupportedInterface,
+    /// Invalid display dimentions
+    ///
+    /// Display dimensions provided in [Builder::display_size] were invalid, e.g. width or height of 0
+    InvalidDisplaySize,
+    /// Invalid display offset.
+    ///
+    /// The active display area, defined by [`display_size`](Builder::display_size) and
+    /// [`display_offset`](Builder::display_offset), extends beyond the boundaries of
+    /// the controller's framebuffer. To resolve this, reduce the offset to a maximum value of
+    /// [`FRAMEBUFFER_SIZE`](Model::FRAMEBUFFER_SIZE) minus [`display_size`](Builder::display_size).
+    InvalidOffset,
```

Comment:
I would suggest to make this consistent with `InvalidDisplaySize` and the corresponding `display_offset` property:
```suggestion
    InvalidDisplayOffset,
```